eth/eventhandler/event_handler.go

```diff
@@ -198,12 +202,12 @@ func (eh *EventHandler) processEvent(txn basedb.Txn, event ethtypes.Log) (Task,
 			eh.logger.Warn("could not parse event",
 				fields.EventName(abiEvent.Name),
 				zap.Error(err))
-			eh.metrics.EventProcessingFailed(abiEvent.Name)
+			eventsProcessFailureCounter.Add(ctx, 1, metric.WithAttributes(eventNameAttribute(abiEvent.Name)))
```

Comment:
Maybe creating a function like:
`RecordFailure(ctx, abiEvent.Name)`
since it repeats itself and we want to make sure all changes will be implemented in all locations.

Reply:
good point. There are quite few places where this metric is being recorded. I added a wrapper function and updated the usages.